vendor/google.golang.org/grpc/clientconn.go

```diff
@@ -950,6 +950,17 @@ func (ac *addrConn) createTransport(connectRetryNum, ridx int, backoffDeadline,
 			Metadata:  addr.Metadata,
 			Authority: ac.cc.authority,
 		}
+		if target.Addr != target.Authority {
+			target.Authority = target.Addr
```

Comment:
Wouldn't it make more sense to follow the same heuristic that was used to determine the Authority when initially creating the `clientconn` rather than just assuming the IP address will do?

i.e. this block: https://github.com/etcd-io/etcd/blob/1ef422c90060ba3e5828304b918e6586eaf81441/vendor/google.golang.org/grpc/clientconn.go#L240-L249

Reply:
Not sure if we can use `cc.parsedTarget.Endpoint` since it was set static initially in `DialContext`. Pretty sure gRPC team can help us find a better way.

I've created an upstream PR https://github.com/grpc/grpc-go/pull/2650, and am still waiting for their feedback.